cmapi/mcs_node_control/models/node_config.py

```diff
@@ -566,4 +566,19 @@ def get_all_dbroots(self, root):
         for i in range(1, mod_count+1):
             for j in range(1, int(smc_node.find(f"./ModuleDBRootCount{i}-3").text) + 1):
                 dbroots.append(smc_node.find(f"./ModuleDBRootID{i}-{j}-3").text)
+
+        # TODO not sure about it
+        if dbroots and self.is_read_only(root):
+            module_logger.warning("Config contains dbroots %s for this read-only node, ignoring", dbroots)
+            return []
+
         return dbroots
+
+    def is_read_only(self, root=None) -> bool:
+        """Checks if this node is in read-only mode"""
+        from cmapi_server.helpers import get_read_only_nodes # Avoid circular import
```

Comment:
We should avoid this circular import "hacks". 
In my practice if any exist - it means that structure is wrong or function in a wrong place.
In the future such a one place produces several another etc.

To be honest some functions from helpers should be here. But not yet.
So my proposal is to move `get_read_only_nodes` function as a method here.